### ISSUE TITLE
Can't convince phpstan about existing offset

### ISSUE THREAD

**Issue by ostrolucky** [phpstan/phpstan-phpunit#53]
```php
self::assertArrayHasKey(1, $ubos);
self::assertFalse($ubos[1]->isActive);
```

produces
```
Offset 1 does not exist on array(MangoPay\Ubo).
```

related class: https://github.com/Mangopay/mangopay2-php-sdk/blob/d967b3fb6f45a0f01891134062265144950c9259/MangoPay/UboDeclaration.php

**Comment by ostrolucky** [comment]
extracting to variable works

**Comment by ntzm** [comment]
As a workaround you can use

```php
$this->assertTrue(array_key_exists('query', $urlParts));
```

**Comment by VincentLanglet** [comment]
I assume you also fixed this with https://github.com/phpstan/phpstan-src/pull/4473 @staabm ?

**Comment by staabm** [comment]
@VincentLanglet yes I think so. as this issue does not has a full reproducer we can't say for sure.

@ostrolucky could you please check whether phpstan-phpunit 2.0.8 in combination with phpstan 2.1.32 still reproduce this problem for you?

**Comment by ostrolucky** [comment]
I can't do it unfortunately since it's a project I no longer has access to. But let's assume it has been fixed, I'll close the issue.
